app/src/main/java/ch/epfl/sweng/SDP/auth/Account.java

```diff
@@ -0,0 +1,338 @@
+     */
+    public static Account getInstance(Context context) {
+        if (instance == null) {
+            createAccount(context, new ConstantsWrapper(), "");
+        }
+
+        return instance;
+    }
+
+    public void setUsersRef(DatabaseReference usersRef) {
+        this.usersRef = usersRef;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public int getTrophies() {
+        return trophies;
+    }
+
+    public void setTrophies(int trophies) {
+        this.trophies = trophies;
+    }
+
+    public int getStars() {
+        return stars;
+    }
+
+    public void setStars(int stars) {
+        this.stars = stars;
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+    public void setUserId(String userId) {
+        this.userId = userId;
+    }
+
+    public String getCurrentLeague() {
+        return currentLeague;
+    }
+
+    public void setCurrentLeague(String currentLeague) {
+        this.currentLeague = currentLeague;
+    }
+
+    /**
+     * Registers this account in Firebase and in the local database.
+     */
+    void registerAccount() throws DatabaseException {
+        usersRef.child(userId).setValue(this, createCompletionListener());
+        localDbHandler.saveAccount(this);
+    }
+
+    /**
+     * Checks in Firebase if username already exists.
+     *
+     * @param newName username to compare
+     * @throws IllegalArgumentException If username is null or already taken
+     * @throws DatabaseException If something went wrong with database.
+     */
+    void checkIfAccountNameIsFree(final String newName)
+            throws IllegalArgumentException, DatabaseException {
+        if (newName == null) {
+            throw new IllegalArgumentException("Username must not be null");
+        }
+        usersRef.orderByChild("username").equalTo(newName)
+                .addListenerForSingleValueEvent(new ValueEventListener() {
+
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot snapshot) {
+                        if (snapshot.exists()) {
+                            throw new IllegalArgumentException("Username already taken.");
+                        }
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError databaseError) {
+                        throw databaseError.toException();
+                    }
+                });
+    }
+
+    /**
+     * Updates username to newName.
+     *
+     * @param newName new username
+     * @throws IllegalArgumentException if username not available anymore
+     * @throws DatabaseException if problems with Firebase
+     */
+    public void updateUsername(final String newName)
+            throws IllegalArgumentException, DatabaseException {
+        checkIfAccountNameIsFree(newName);
```

Comment:
This has nothing to do with what you're doing, but i just saw that this call to checkIfAccountNameIsFree is useless since the method runs asynchronously and will not return until the following code has already been executed.
If you have time, try to change this. Otherwise we will fix that in a later PR.